BetaUtils.Core/Exceptions/Middleware/TrackedExceptions.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+
+namespace BetaUtils.Core.Exceptions.Middleware;
+
+public static class TrackedExceptions
+{
+    private static Dictionary<Type, int> _exceptions = DefaultExceptions();
+
+    /// <summary>
+    /// Add exception with status code to tracked exceptions. Only one exception of same type may exist in dictionary.
+    /// </summary>
+    /// <returns>True if exception added successfully.</returns>
+    public static bool Add<TException>(int statusCode) where TException : Exception
```

Comment:
System.Net.HttpStatusCode in my opinion is more generic and requires fraction more effort to follow up since each of the enum naming does not have status code in name, so having client demanding instead of 200 OK to return 207 (for some reason...) would require for you to either go to google and search name of the status code, or to open microsoft lib and search for the code while in `Microsoft.AspNetCore.Http.StatusCodes` name of the status code is already included.

Another point is that status codes are numbers and .net handles them as such. In order to use System.Net.HttpStatusCode you would have to cast that enum to int either directly of safely and i am not sure if that is simpler or nicer. 😅